src/auto/output_stream.rs

```diff
@@ -0,0 +1,304 @@
+// This file was generated by gir (d8a605d) from gir-files (469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+use InputStream;
+use OutputStreamSpliceFlags;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct OutputStream(Object<ffi::GOutputStream, ffi::GOutputStreamClass>);
+
+    match fn {
+        get_type => || ffi::g_output_stream_get_type(),
+    }
+}
+
+pub trait OutputStreamExt {
+    fn clear_pending(&self);
+
+    fn close<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P) -> Result<(), Error>;
+
+    fn close_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, io_priority: i32, cancellable: P, callback: Q);
+
+    fn flush<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P) -> Result<(), Error>;
+
+    fn flush_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, io_priority: i32, cancellable: P, callback: Q);
+
+    fn has_pending(&self) -> bool;
+
+    fn is_closed(&self) -> bool;
+
+    fn is_closing(&self) -> bool;
+
+    //#[cfg(any(feature = "v2_40", feature = "dox"))]
+    //fn printf<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P, error: &mut Error, format: &str, : /*Unknown conversion*//*Unimplemented*/Fundamental: VarArgs) -> Option<usize>;
+
+    fn set_pending(&self) -> Result<(), Error>;
+
+    fn splice<'a, P: IsA<InputStream>, Q: Into<Option<&'a Cancellable>>>(&self, source: &P, flags: OutputStreamSpliceFlags, cancellable: Q) -> Result<isize, Error>;
+
+    fn splice_async<'a, P: IsA<InputStream>, Q: Into<Option<&'a Cancellable>>, R: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, source: &P, flags: OutputStreamSpliceFlags, io_priority: i32, cancellable: Q, callback: R);
+
+    //#[cfg(any(feature = "v2_40", feature = "dox"))]
+    //fn vprintf<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P, error: &mut Error, format: &str, args: /*Unknown conversion*//*Unimplemented*/Unsupported) -> Option<usize>;
+
+    fn write<'a, P: Into<Option<&'a Cancellable>>>(&self, buffer: &[u8], cancellable: P) -> Result<isize, Error>;
+
+    fn write_all<'a, P: Into<Option<&'a Cancellable>>>(&self, buffer: &[u8], cancellable: P) -> Result<usize, Error>;
+
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    fn write_all_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<usize, Error>) + Send + Sync + 'static>(&self, buffer: &[u8], io_priority: i32, cancellable: P, callback: Q);
+
+    fn write_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, buffer: &[u8], io_priority: i32, cancellable: P, callback: Q);
```

Comment:
Buffer must stay alive until the async callback (or rather the finish method) is called